libs/libc/time/lib_localtime.c

```diff
@@ -307,14 +307,15 @@ struct rule_s
 
 /* The minimum and maximum finite time values.  */
 
-static const time_t g_min_timet =
-  (TYPE_SIGNED(time_t)
-    ? (time_t)-1 << (CHAR_BIT * sizeof(time_t) - 1)
+static const int_fast64_t g_min_timet =
+  (TYPE_SIGNED(int_fast64_t)
+    ? (int_fast64_t)1 << (CHAR_BIT * sizeof(int_fast64_t) - 1)
     : 0);
 
-static const time_t g_max_timet =
-  (TYPE_SIGNED(time_t)
-    ? - (~ 0 < 0) - ((time_t)-1 << (CHAR_BIT * sizeof(time_t) - 1))
+static const int_fast64_t g_max_timet =
+  (TYPE_SIGNED(int_fast64_t)
+    ? - (~ (int_fast64_t)0 < 0) - ((int_fast64_t)1 <<
+                                   (CHAR_BIT * sizeof(int_fast64_t) - 1))
```

Comment:
I think that we need to switch to a fixed values here since `int_fast64_t` is used. The initial implementation was relying on assumption that `time_t` could be either signed or unsigned. If we switch to a fixed type then all this calculations do not make sense.